gremlin-scala/src/main/scala/gremlin/scala/ScalaGraph.scala

```diff
@@ -6,21 +6,25 @@ import shapeless._
 
 case class ScalaGraph[G <: Graph](graph: G) {
 
-  def addVertex() = ScalaVertex(graph.addVertex())
+  def addVertex(): ScalaVertex = graph.addVertex()
 
-  def addVertex(label: String) = ScalaVertex(graph.addVertex(label))
+  def addVertex(label: String): ScalaVertex = graph.addVertex(label)
 
   def addVertex(properties: Map[String, Any]): ScalaVertex = {
-    val v = addVertex()
-    v.setProperties(properties)
-    v
+    graph.addVertex(properties.toSeq.flatMap {
+      case (k, v) =>
+      val label = T.label.name
+      val id = T.id.name
+      k match {
+        case `label` => Seq(T.label, v.asInstanceOf[AnyRef])
+        case `id` => Seq(T.id, v.asInstanceOf[AnyRef])
+        case _ => Seq(k, v.asInstanceOf[AnyRef])
+      }
+    }: _*)
   }
```

Comment:
HMap does not  support the map and flatMap functions needed to convert into the varargs required in the Java API.